path_roles.go

```diff
@@ -51,6 +55,8 @@ type AzureRole struct {
 	RoleName string `json:"role_name"` // e.g. Owner
 	RoleID   string `json:"role_id"`   // e.g. /subscriptions/e0a207b2-.../providers/Microsoft.Authorization/roleDefinitions/de139f84-...
 	Scope    string `json:"scope"`     // e.g. /subscriptions/e0a207b2-...
+
+	RoleAssignmentID string `json:"role_assignment_id"` // e.g. /subscriptions/e0a207b2-.../providers/Microsoft.Authorization/roleAssignments/de139f84-...
```

Comment:
We have to store the RoleAssignmentID somewhere because the Azure client only offers a way to delete RoleAssignments by ID and Name (name is effectively the short ID form). This seems like the best place to store the RoleAssignmentID, however, it also means that `vault read azure/roles/:role` will return the `RoleAssignmentID`. I'm not sure that is a bad thing, being able to see what the exact RoleAssignment is can be useful I think. Alternatively, I can update https://github.com/DataDog/vault-plugin-secrets-azure/blob/21b978d8418cf7791a68f6473992f798e244c4f8/path_roles.go#L416 to strip the `role_assignment_id` and we can always come back later and add it.

Reply:
Can RoleAssignmentID not be determined at revoke time? Or does Azure support creation of multiple assignments of the same role/scope/principal combination?

What's the impact to existing role assignments not tracked in this field?

Reply:
Its a good question. It doesn't look like you can create multiple RoleAssignments with the same RoleDefinitionID, Scope, Principal (on the UI it is an error, the CLI gives the same response for multiple requests).

The downside is that we'd have to probably call `ListForScope` which is paginated. So it's at least 2 calls to Azure (one lookup/filter, one delete by RoleAssignmentID) but over time this is going to grow as the number of RoleAssignments go up and the list has to be paginated.

However, for existing roles, the downside is that the `role_assignment_id` is not tracked and therefore can result in RoleAssignments which do not get removed correctly (see the screenshot in an earlier comment). Maybe it would be better to only do a list/filter if `role_assignment_id` does not exist?

Reply:
> Maybe it would be better to only do a list/filter if role_assignment_id does not exist?

This seems viable, although it might not be worth the added complexity.

Reply:
> This seems like the best place to store the RoleAssignmentID, however, it also means that vault read azure/roles/:role will return the RoleAssignmentID. I'm not sure that is a bad thing, being able to see what the exact RoleAssignment is can be useful I think.

I ended up hiding `role_assignment_id` from the response of `azure/roles/:name` read requests.